RecoMET/METFilters/plugins/BadChargedCandidateSummer16Filter.cc

```diff
@@ -0,0 +1,141 @@
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+
+//
+// class declaration
+//
+
+class BadChargedCandidateSummer16Filter : public edm::global::EDFilter<> {
+public:
+  explicit BadChargedCandidateSummer16Filter(const edm::ParameterSet&);
+  ~BadChargedCandidateSummer16Filter();
+
+private:
+  virtual bool filter(edm::StreamID iID, edm::Event&, const edm::EventSetup&) const override;
+
+      // ----------member data ---------------------------
+
+  edm::EDGetTokenT<edm::View<reco::Candidate> >   tokenPFCandidates_;
+  edm::EDGetTokenT<edm::View<reco::Muon> >   tokenMuons_;
+
+  const bool taggingMode_;
+  const bool debug_;
+  const double          maxDR_;
+  const double          minPtDiffRel_;
+  const double          minMuonTrackRelErr_;
+  const double          minMuonPt_;
+
+};
+
+//
+// constructors and destructor
+//
+BadChargedCandidateSummer16Filter::BadChargedCandidateSummer16Filter(const edm::ParameterSet& iConfig)
```

Comment:
given similarities with BadChargedCandidateFilter, the code should be refactored

see comments from Sept 2016
https://github.com/cms-sw/cmssw/pull/15761#issuecomment-245742994